File: Data Structures & Algorithms/longest-consecutive-sequence/submission-9.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        
        # idea: use a set to see if a sequence exits
        # instead of climing up, we can climb down

        seq = set()
        maxCount = 0
        for i in nums:
            count = 1
            # down
            j = i - 1
            while j in seq:
                count += 1
                j -= 1
            # up
            j = i + 1
            while j in seq:
                count += 1
                j += 1
            maxCount = max(count, maxCount)
            seq.add(i)
        
        return maxCount

File: Data Structures & Algorithms/two-integer-sum/submission-9.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int, int> hmap;

        for (int i = 0; i < nums.size(); i++){
            if (hmap.count(target - nums[i])) {
                return {min(i, hmap[target - nums[i]]), 
                        max(i, hmap[target - nums[i]])};
            }
            hmap[nums[i]] = i;
        }

        return {0, 1};
    }
};
